app/order-history/page.tsx

```diff
@@ -0,0 +1,381 @@
+      setStatusFilter(value);
+    } else if (paymentStatuses.includes(value)) {
+      setPaymentStatusFilter(value);
+    }
+  };
+
+  // Clear all filters
+  const clearFilters = () => {
+    setFilter("");
+    setServiceFilter([]);
+    setLogisticsFilter(null);
+    setStatusFilter(null);
+    setPaymentStatusFilter(null);
+  };
+
+  // Memoized filter
+  const filteredOrders = useMemo(() => {
+    return orderData.filter(order => {
+      const matchesName = order.name.toLowerCase().includes(filter.toLowerCase());
+      const matchesServiceType = serviceFilter.length === 0 || serviceFilter.includes(order.serviceType);
+      const matchesLogistics = !logisticsFilter || order.logistics === logisticsFilter;
+      const matchesStatus = !statusFilter || order.status === statusFilter;
+      const matchesPaymentStatus = !paymentStatusFilter || order.paymentStatus === paymentStatusFilter;
+
+      return matchesName && matchesServiceType && matchesLogistics && matchesStatus && matchesPaymentStatus;
+    });
+  }, [filter, serviceFilter, logisticsFilter, statusFilter, paymentStatusFilter]);
+
+  return (
+    <div className="p-6 bg-gray-50 min-h-screen">
+      <h1 className="text-2xl font-bold mb-6 text-gray-800">Order History</h1>
+
+      {/* Search and filter dropdown container */}
+      <div className="flex items-center mb-6 gap-4">
+        <Input
+          placeholder="Search"
+          value={filter}
+          onChange={(e) => setFilter(e.target.value)}
+          className="border border-gray-300 px-3 py-2 rounded-lg shadow-sm w-full"
+        />
+
+        <div className="relative">
+          <select 
+            onChange={handleFilterChange}
+            className="border border-gray-300 px-3 py-2 rounded-lg shadow-sm w-full"
+            style={{ backgroundColor: '#DBEAFF', color: 'black' }} 
+          >
+            {/* Dropdown for filters */}
+            <option value="" className="text-black">Select Filter</option>
+            <optgroup label="Service Types">
+              {serviceTypes.map((type) => (
+                <option key={type} value={type} className="text-black">{type}</option>
+              ))}
+            </optgroup>
+            <optgroup label="Logistics">
+              {logisticsOptions.map((logistics) => (
+                <option key={logistics} value={logistics} className="text-black">{logistics}</option>
+              ))}
+            </optgroup>
+            <optgroup label="Order Statuses">
+              {statuses.map((status) => (
+                <option key={status} value={status} className="text-black">{status}</option>
+              ))}
+            </optgroup>
+            <optgroup label="Payment Statuses">
+              {paymentStatuses.map((paymentStatus) => (
+                <option key={paymentStatus} value={paymentStatus} className="text-black">{paymentStatus}</option>
+              ))}
+            </optgroup>
+          </select>
+        </div>
+      </div>
```

Comment:
seems like manually mo gin create ang dropdown? i believe may shadcn component for this. but we can keep this, just extract this to another file para componentized siya.

probably something like `SearchFilterDropdown.tsx` nga located inside the same folder lang.

React's power is to componentized and isolate different features para we don't have to think how each feature or section works.